test/katex-spec.js

```diff
@@ -1042,6 +1042,16 @@ describe("A left/right parser", function() {
         var normalEmpty = "\\Bigl .";
         expect(normalEmpty).toParse();
     });
+
+    it("should handle \\middle", function() {
+        var normalMiddle = "\\left( \\dfrac{x}{y} \\middle| \\dfrac{y}{z} \\right)";
+        expect(normalMiddle).toParse();
+    });
+
+    it("should error when \\middle is not in \\left...\\right", function() {
+        var unmatchedMiddle = "(\\middle|\\dfrac{x}{y})";
```

Comment:
Well, the `\middle` won't be rewritten with the correct delimiter if this happens, because it won't be a direct child of the `\left\right` thing, so it probably shouldn't work? (That's actually just a guess, you should test it)